### PR TITLE
fixes #715 Added Append Method for BytesBuilder to take byte[], offset, and length.

### LINKED ISSUE #715: Add append method for ByteBuilder that takes a byte array, length, and offset
BytesBuilder has a method that takes a byte array, but not one that takes a part of a byte array.

### PR DESCRIPTION
**Pull request by cjmctague**
- Added Append Method for BytesBuilder to take byte[], offset, and length.
- Added unit test to test method.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java [hunk 1]
```diff
@@ -71,6 +71,17 @@ public void testSingleByte() {
   }
 
   @Test
+  public void testArraySection() {
+    BytesBuilder bb = Bytes.newBuilder();
+
+    bb.append(new byte[] {'c', (byte) 0, (byte) 127, (byte) 128, (byte) 255}, 0, 3);
+
+    Bytes bytes = bb.toBytes();
```
Comment (mikewalch) [inline modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java]: This test looks broken.  The `assertEquals` should verify the data in `bytes`.

**Comment by cjmctague** [comment]
@mikewalch should be fixed now

File: modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java [hunk 2]
```diff
@@ -71,6 +71,17 @@ public void testSingleByte() {
...
+    Assert.assertEquals(Bytes.of(new byte[] {'c', (byte) 0, (byte) 127, (byte) 128, (byte) 255}),
```
Comment (mikewalch) [inline modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java]: I ran the test in my IDE and it's still failing. The 'bytes' object only contains a 3 byte section of the original array that was passed to append method.  The `assertEquals` method should confirm this.

Comment (keith-turner) [inline modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java]: Could call the `Bytes.of(byte[],int offset, int len)` method to produce the expected data.

File: modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java [hunk 3]
```diff
@@ -71,6 +71,17 @@ public void testSingleByte() {
   }
 
   @Test
+  public void testArraySection() {
```
Comment (keith-turner) [inline modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java]: The following additional test would be nice :
 * calling append w/ a non-zero offset
 * calling this new append method multiple times on a single BytesBuilder

File: modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java [hunk 4]
```diff
@@ -71,6 +71,17 @@ public void testSingleByte() {
...
+    BytesBuilder bb = Bytes.newBuilder();
+
+    bb.append(new byte[] {'c', (byte) 0, (byte) 127, (byte) 128, (byte) 255}, 0, 3);
```
Comment (keith-turner) [inline modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java]: could keep a reference to this byte array and reuse it to make lines shorter.

**Comment by keith-turner** [comment]
@cjmctague  are you squashing and force pushing?  If so we usually avoid this while code review is in progress, in favor of only appending commits.  This makes it easier for the reviewer to see whats changed.  After the code review is finished we may squash it into one commit.

File: modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java (hunk 3 above)
Comment (keith-turner) [inline modules/core/src/test/java/org/apache/fluo/core/data/BytesBuilderTest.java]: The test looks much better.  Thanks for making those changes.

**Comment by keith-turner** [comment]
+1

**Comment by keith-turner** [comment]
@cjmctague  would you like to be added to the [people page](http://fluo.apache.org/people/) as a contributor?  If so, what information would you like to put there?

**Comment by keith-turner** [comment]
@cjmctague For future reference, the commit message should have had `fixes #715` in it.  Maybe one of your previous commit messages had that. I see it in the pull request title, but not the final commit.

**Pull request merged**